device/core/src/internal_client.ts

```diff
@@ -337,6 +339,68 @@ export abstract class InternalClient extends EventEmitter {
     }, abandonCallback);
   }
 
+  /**
+   * Sends a property update patch to the Azure IoT Hub or Azure IoT Edge Hub service.
+   *
+   * @param {ClientPropertyCollection} propertyCollection A ClientPropertyCollection object representing the property patch.
+   * @param {(err: Error) => void}     done               A callback used to get notified of the success or failure of the operation.
+   *                                                      If no callback is specified, a promise is returned instead.
+   */
+  updateClientProperties(propertyCollection: ClientPropertyCollection): Promise<void>;
+  updateClientProperties(propertyCollection: ClientPropertyCollection, done: (err: Error) => void): void;
+  updateClientProperties(propertyCollection: ClientPropertyCollection, done?: (err: Error) => void): Promise<void> | void {
+    return callbackToPromise((_callback) => {
+      this.getTwin((err, twin) => {
+        if (err) {
+          _callback(err);
+          return;
+        }
+        (twin as any)._updateReportedProperties(propertyCollection.backingObject, _callback);
```

Comment:
do we have a type for twin? avoid using `any`

Reply:
It's done this way because `_updateReportedProperties` is actually a private member of `Twin` and tsc would complain about it if you took out the `as any`. 

We could have called the `properties.reported.update` function instead, but this is a good opportunity to get rid of the [limitation where having a reported property called `update` would break things](https://github.com/Azure/azure-iot-sdk-node/issues/578). 

And IMO we should keep it private because we expose `Twin` directly to the user with the "legacy" API, and we don't want user's calling that directly. 

I'll add a comment explaining why the `as any` is there.